app/actions.js

```diff
@@ -439,25 +443,27 @@ export const loadFromDisk = () => async dispatch => {
     state = {}
   }
   const stateKeys = Object.keys(state)
-  for (const key of stateKeys) {
-    const cabalState = JSON.parse(state[key])
-    dispatch(addCabal(cabalState))
-  }
+  stateKeys.forEach((key) => {
+    const { addr, settings } = JSON.parse(state[key])
+    dispatch(restoreCabalSettings({ addr, settings }))
+  })
+  stateKeys.forEach((key) => {
+    const { addr, settings } = JSON.parse(state[key])
+    dispatch(addCabal({ addr, settings }))
+  })
```

Comment:
is there a reason for the duplicated looping of `stateKeys`? could be nice to add a comment in that case to document the gotcha. otherwise toss `dispatch(addCabal({ addr, settings }))` into the first loop :)

Reply:
Yep! We need to load in all of the settings for all cabals into the store first before we start initializing cabals. `addCabal` kicks off all kinds of things and its easier/saner to have the state of the app from last time loaded before that starts.

Also! After I made the PR i read through the changes again and this did stand out to me that i did two similar loops in a row. THEN i realized that for some reason we stringify the save data twice and im not sure why. I thought about simplifying that, but it would break everyone's existing save files. So i left it as is so i'd come back later and solve it. There's other changes i'd like to make to how the save file is structured and i think in a future release we could have a migration function to make that seamless for people.